docs/spec_draft.md

```diff
@@ -1810,6 +1811,83 @@ where `i[d] = j[permutation[d]]`.
 
 [Back to Ops](#index-of-ops)
 
+## stablehlo.triangular_solve
+
+### Semantics
+
+Solves systems of linear equations of a batch of matrices with lower or upper
+triangular coefficients by forward- or back-substitution.
+
+More formally, for all `i`, `result[i0, ..., iR-3, :, :]` is the solution to
+`op(a[i0, ..., iR-3, :, :]) * x = b[i0, ..., iR-3, :, :]` (`left_side` is true)
+`x * op(a[i0, ..., iR-3, :, :]) = b[i0, ..., iR-3, :, :]` (`left_side` is false)
+solving for the variable `x` given `a` and `b`, where `op(a)` is either
+`op(a) = a`, or `op(a) = Transpose(a)`, or `op(a) = Conj(Transpose(a))`. The
+output values in the opposite triangle, i.e. the strict upper triangle or strict
+lower triangle correspondingly, are implementation-defined.
+
+If `unit_diagonal` is true, the diamonal elements of `a` are assumed to be 1 and
+not accessed.
+
+`transpose_a` can be one of the following:
+  * `TRANSPOSE_INVALID`: Illegal matrix; transpose is invalid.
```

Comment:
This option lead to verification error both in MHLO and HLO?
https://github.com/openxla/stablehlo/blob/ca7ef4f80152e9d5314e4c9750590d018553e91c/stablehlo/dialect/TypeInference.cpp#L1354

We should add that in Constraints.
Next, if that is not a valid value can we know why we have this option in the first place.

Reply:
The reason we have this check is that proto in HLO has enum value of 0 as default. As for why NO_TRANSPOSE is not the default of 0, I do not know. To sync with HLO, this enum value seems necessary. @burmako Thoughts?
https://github.com/tensorflow/tensorflow/blob/910db383e2d5a649eda3ddf3e474852e3a6fddbb/tensorflow/compiler/xla/xla_data.proto#L670-L675

Reply:
Let us drop the TRANSPOSE_INVALID.